pkg/api/openshiftcluster.go

```diff
@@ -122,6 +123,8 @@ type OpenShiftClusterProperties struct {
 
 	ServicePrincipalProfile ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
 
+	PlatformWorkloadIdentityProfile PlatformWorkloadIdentityProfile `json:"platformWorkloadIdentityProfile,omitempty"`
```

Comment:
This field is a pointer in the design doc to allow for it to be `nil` for CSP clusters.